src/python/grpcio_health_checking/grpc_health/v1/_async.py

```diff
@@ -84,9 +84,10 @@ async def _set(
     ) -> None:
         if service in self._server_watchers:
             condition = self._server_watchers.get(service)
-            async with condition:
-                self._server_status[service] = status
-                condition.notify_all()
+            if condition is not None:
```

Comment:
Is this because the `get` method of `MutableMapping` may return `None`? `collections.defaultdict[str, asyncio.Condition]` will never return `None`, so we're sort of guarding against an impossible condition here.

Can we put a better type annotation on `_server_watchers` to make this issue go away?

Reply:
Done